prov/verbs/src/verbs_eq.c

```diff
@@ -752,11 +752,81 @@ static size_t fi_ibv_eq_read_event(struct fi_ibv_eq *eq, uint32_t *event,
 	return ret;
 }
 
+static int
+fi_ibv_read_async_event(struct fi_ibv_domain *domain,
+                        struct fi_ibv_eq *eq)
+{
+	struct ibv_async_event async_event;
+	int ret = 0;
+	int error = 0;
+
+	if (ibv_get_async_event(domain->verbs, &async_event) != 0)
+		return 0;
+
+	switch (async_event.event_type) {
+	/* Errors should be reported */
+	case IBV_EVENT_CQ_ERR: /* CQ is in error (CQ overrun) */
+		error = -FI_EOVERRUN;
+		break;
+
+	case IBV_EVENT_SRQ_ERR: /* Error occurred on an SRQ */
+	/* fallthrough */
+	case IBV_EVENT_PORT_ERR: /* Link became unavailable on a port */
+	/* fallthrough */
+	case IBV_EVENT_PATH_MIG_ERR: /* A connection failed to migrate to the alternate path */
+	/* fallthrough */
+	case IBV_EVENT_DEVICE_FATAL: /* CA is in FATAL state */
+		error = -FI_EFAULT;
+		break;
+
+	/* Ignored events */
+	case IBV_EVENT_COMM_EST: /* Communication was established on a QP */
+	/* fallthrough */
+	case IBV_EVENT_SQ_DRAINED: /* Send Queue was drained of outstanding messages in progress */
+	/* fallthrough */
+	case IBV_EVENT_PATH_MIG: /* A connection has migrated to the alternate path */
+	/* fallthrough */
+	case IBV_EVENT_QP_LAST_WQE_REACHED: /* Last WQE Reached on a QP associated with an SRQ */
+	/* fallthrough */
+	case IBV_EVENT_SRQ_LIMIT_REACHED: /* SRQ limit was reached */
+	/* fallthrough */
+	case IBV_EVENT_PORT_ACTIVE: /* Link became active on a port */
+	/* fallthrough */
+	case IBV_EVENT_LID_CHANGE: /* LID was changed on a port */
+	/* fallthrough */
+	case IBV_EVENT_PKEY_CHANGE: /* P_Key table was changed on a port */
+	/* fallthrough */
+	case IBV_EVENT_SM_CHANGE: /* SM was changed on a port */
+	/* fallthrough */
+	case IBV_EVENT_CLIENT_REREGISTER: /* SM sent a CLIENT_REREGISTER request to a port */
+	/* fallthrough */
+	case IBV_EVENT_GID_CHANGE: /* GID table was changed on a port */
+	/* fallthrough */
+	default:
+		ret = 0;
+		goto ack_async_event;
+	}
+
+	fastlock_acquire(&eq->lock);
+	eq->err.err = error;
+	eq->err.fid = &domain->util_domain.domain_fid.fid;
+	eq->err.prov_errno = async_event.event_type;
+	fastlock_release(&eq->lock);
```

Comment:
It might be useful to have a debug message for the event being reported.

Reply:
Thanks, I have added a debug message.